templates/vault/0/docker-compose.yml.tpl

```diff
@@ -0,0 +1,38 @@
+version: '2'
+volumes:
+  vault-config:
+    driver: ${VOLUME_DRIVER}
+  vault-file:
+    driver: ${VOLUME_DRIVER}
+services:
+  vault-lb:
+    image: rancher/lb-service-haproxy:v0.9.1
+    ports:
+    - ${VAULT_LISTEN_PORT}:8200/tcp
+    - ${VAULT_CLUSTER_PORT}:8201/tcp
+    labels:
+      io.rancher.container.agent.role: environmentAdmin,agent
+      io.rancher.container.agent_service.drain_provider: 'true'
+      io.rancher.container.create_agent: 'true'
+{{- if .Values.HOST_LABEL }}
+      io.rancher.scheduler.affinity:host_label: ${HOST_LABEL}
+{{- end }}
+  vault:
+    image: vault:0.9.6
+    cap_add:
+    - IPC_LOCK
+{{- if .Values.VAULT_BACKEND }}
+    external_links:
+    - ${BACKEND_SERVICE}:SERVICE
+{{- end }}
+    environment:
+      VAULT_REDIRECT_INTERFACE: "eth0"
+      VAULT_CLUSTER_INTERFACE: "eth0"
+      VAULT_LOCAL_CONFIG: |
+        { "storage":{"{{.Values.VAULT_BACKEND}}":{ {{.Values.BACKEND_CONFIGURATION}} }},"listener":{"tcp":{"address":"0.0.0.0:8200","tls_disable":1}},"cluster_name":"{{.Values.VAULT_CLUSTER_NAME}}"}
```

Comment:
Have you tested it?? This value is not working, generating error when deploy, (tested with all options) ` Error (yaml: line 26: did not find expected key)`

Working fine with this. Could you please update it??
```
-        { "storage":{"{{.Values.VAULT_BACKEND}}":{ {{.Values.BACKEND_CONFIGURATION}} }},"listener":{"tcp":{"address":"0.0.0.0:8200","tls_disable":1}},"cluster_name":"{{.Values.VAULT_CLUSTER_NAME}}"}
+        { 
+          "storage": {"${VAULT_BACKEND}": {${BACKEND_CONFIGURATION}}},
+          "listener": {"tcp": {"address": "0.0.0.0:8200", "tls_disable": 1}},
+          "cluster_name": "${VAULT_CLUSTER_NAME}" 
+        }
```

Reply:
Yes, I tested it.  But, I can replicate what you are seeing...  if you tab through all the fields on the form (even without changing anything), it works.  But, if you don't tab through the BACKEND_CONFIGURATION field, it fails with the error.  Does Rancher do something on focus or on lost focus that maybe sets a field value?   Is there any way of dumping out the generated YAML that gets produced to try and help track down where this could be coming from?

Reply:
With the changes i've requested, works with all options, changing whatever field. The problem seems to be on line 24 if statement and in `VAULT_LOCAL_CONFIG` definition.
Please, try applying changes requested.

Reply:
I'm not sure how to handle this... even after making your requested change, I still see the same behavior:  If you select the Catalog item and leave everything at default, it fails.  But, if you click in the BACKEND_CONFIGURATION field, don't change anything, and tab out, it works just the way you expect it to work.

Reply:
mmm..what strange...For me is working now...Without changing anything, just selecting package and deploy... Using last stable v1.6.16...
But i think i got the error, my fault btw...It's on rancher-compose line 62 and 63... Default value is defined as it's a multiline type and add a `\n` to the string. 
Please change it to the following and tell me if works for you...
```
-      default: |
-        "path":"/vault/file"
+      default: '"path":"/vault/file"'
```